python/ray/data/_internal/planner/sort.py

```diff
@@ -0,0 +1,65 @@
+from functools import partial
+from typing import List, Tuple
+
+from ray.data._internal.execution.interfaces import (
+    AllToAllTransformFn,
+    RefBundle,
+    TaskContext,
+)
+from ray.data._internal.planner.exchange.push_based_shuffle_task_scheduler import (
+    PushBasedShuffleTaskScheduler,
+)
+from ray.data._internal.planner.exchange.pull_based_shuffle_task_scheduler import (
+    PullBasedShuffleTaskScheduler,
+)
+from ray.data._internal.planner.exchange.sort_task_spec import SortKeyT, SortTaskSpec
+from ray.data._internal.stats import StatsDict
+from ray.data.context import DatasetContext
+
+
+def generate_sort_fn(
+    key: SortKeyT,
+    descending: bool,
+) -> AllToAllTransformFn:
+    """Generate function to sort blocks by the specified key column or key function."""
+    # TODO: validate key with block._validate_key_fn.
```

Comment:
Created https://github.com/ray-project/ray/issues/32137 for followup. This needs more refactoring, as it involves `Dataset` object.